odoo/addons/base/models/res_users.py

```diff
@@ -699,7 +699,7 @@ def context_get(self):
 
     @tools.ormcache('self._uid')
     def _get_company_ids(self):
-        return frozenset(self.company_ids.ids)
+        return frozenset(self.env['res.company'].search([('user_ids', 'in', self._uid)]).ids)
```

Comment:
That's a change of behavior: returning the companies of `self._uid` instead of `self.id`. The calls seems to often be `user._get_company_ids()` and meant to replace `user.company_ids.ids`, so let's keep the previous behavior. 

Of course the ormcache key seems to make little sense, which is probably what caused the confusion. We should fix it as well to be `self.id`.

```suggestion
        return frozenset(self.env['res.company'].search([('user_ids', 'in', self.id)]).ids)
```


+1 also for forcing `active_test` for consistent values. Note that instead of forcing it, you could also pass explicitly `[('active', '=', True)]` in the domain.

Reply:
One thing to note too: half of the 6 extra queries in the tests were artifacts of the test warmup/multi-user-run, with a cache miss happening later instead of sooner. In other words, the total number of queries for the whole test (including warmup) increased by 3 only.

Using a `search()` inside `_get_company_ids()` is similar: `user.company_ids` is not set anymore, possibly causing an extra query later when that value is used.

Anyway, it doesn't matter, let's go for that if it reduces the diff -- I'm still convinced that this is completely insignificant because those queries are very cheap.